src/lib.rs

```diff
@@ -314,20 +314,29 @@ impl<T: LedgerAccess, const N: usize> Ledger<T, N> {
             let record_start = self.records[index].region.start;
             let record_end = self.records[index].region.end;
 
-            if region.end < record_start || region.start > record_end {
+            if region.start >= record_end {
                 // Skip:
+                // [   ]   XXXXX
```

Comment:
These comments are cute, but they don't explain why some of these cases cause `continue` while others cause `return`.

Reply:
added more

Reply:
and change the control flow to a match statement, so we can catch the `unreachable!()` on logic error.